--- ipaddr.old.js	2015-12-29 00:25:10.896307421 +0100
+++ ipaddr.js	2015-12-29 00:31:11.003092184 +0100
@@ -192,7 +192,7 @@
         return compactStringParts.push(part);
       };
       state = 0;
-      for (_i = 0, _len = stringParts.length; _i < _len; _i++) {
+      for (_i = stringParts.length - 1, _len = 0; _i >= _len; _i--) {
         part = stringParts[_i];
         switch (state) {
           case 0:
@@ -225,7 +225,7 @@
         pushPart('');
         pushPart('');
       }
-      return compactStringParts.join(":");
+      return compactStringParts.reverse().join(":");
     };
 
     IPv6.prototype.toByteArray = function() {
